contracts/src/v0.8/functions/dev/1_0_0/FunctionsCoordinator.sol

```diff
@@ -248,6 +246,8 @@ contract FunctionsCoordinator is OCR2Base, IFunctionsCoordinator, FunctionsBilli
       } else if (result == FulfillResult.INSUFFICIENT_SUBSCRIPTION_BALANCE) {
         emit InsufficientSubscriptionBalance(requestIds[i]);
       }
+
+      emit OracleResponse(requestIds[i], msg.sender);
```

Comment:
I think this should happen every time EXCEPT when we got INVALID_REQUEST_ID because that is not a fulfillment.

The definition of fulfillment for the node should be: removed an existing requestID from s_requestCommitments

Reply:
Ah, well then how we had it before is correct.
The request is only fulfilled on status code 0 or 1

Reply:
Are you sure? What about INSUFFICIENT_GAS? In that case we still remove from s_requestCommitments and we should emit OracleResponse.

Reply:
Why would the NOP not providing enough gas count as a fulfillment?
If we remove the request from a pending state here the user will not be able to time the request out

Reply:
OK maybe "fulfillment" is the wrong term here.

I'm looking at _fulfillAndBill(). First thing that happens is `delete s_requestCommitments[requestId];`.
This can happen only once for each request because next time we'll get INVALID_REQUEST_ID. 

Nodes care only about that thing happening and we should have an event to reflect that (removing from s_requestCommitments). Whatever happens next (success, not enough gas, etc) doesn't matter for the node.

Reply:
This seems like a problem:
1. A faulty node attempts to call the user's callback, but does not provide enough remaining gas for the user's callbackGasLimit
2. This causes a INSUFFICENT_GAS error, which means that a non-faulty node should re-attempt to call the user's callback.  However, this is not possible since we have deleted the request commitment.

Reply:
Also, since an `OracleResponse` event was emitted even though the DON failed to call the user's callback, the nodes will think that the request was successfully completed even though it was not.

Reply:
Therefore, we should emit an `OracleResponse` in all cases except for INVALID_REQUEST_ID and INSUFFICIENT_GAS.

However, this still doesn't fix the problem of prematurely deleting the request commitment in the case where a faulty node provided insufficient gas.  This must still be addressed before I can approve.

Reply:
Yeah this needs some more work to define all paths precisely.